src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -707,6 +724,29 @@ export class CellModel implements ICellModel {
 		return source;
 	}
 
+	// Run ADS commands via a notebook. Configured by a setting (turned off by default).
+	private checkForAdsCommandMagic(): boolean {
+		if (!this._isCommandExecutionSettingEnabled) {
+			return false;
+		}
+
+		let executeCommandMagic = '%%AZDATA_EXECUTE_COMMAND';
+		if (this._source && this._source.length) {
+			if (this._source[0].startsWith(executeCommandMagic)) {
+				let commandNamePlusArgs = this._source[0].replace(executeCommandMagic + ' ', '');
+				if (commandNamePlusArgs) {
+					let commandName = commandNamePlusArgs.split(' ')[0];
+					if (commandName) {
+						let args = commandNamePlusArgs.replace(commandName + ' ', '');
+						this._commandService.executeCommand(commandName, [args]);
```

Comment:
This is going to pass all the args as a single string - we should be splitting them up. But when you do so make sure we handle quoted args correctly as well...there might be a library that would help with this?